gpxpy/parser.py

```diff
@@ -165,7 +165,7 @@ def __init__(self, xml_or_file=None, parser=None):
 
     def init(self, xml_or_file):
         text = xml_or_file.read() if hasattr(xml_or_file, 'read') else xml_or_file
-        if text[:3] == "\xEF\xBB\xBF": #Remove utf-8 Byte Order Mark (BOM) if present
+        if text.startswith(u'\xEF\xBB\xBF'):  # Remove utf-8 Byte Order Mark (BOM) if present
```

Comment:
I believe only @tkrajina understands what's supposed to happen here. 

From what I see, it can either accept a file or a string-like object. A file.read() is bytes in Python 2 and unicode in Python 3. But when it's a string-like object it can be anything in both Py 2 and 3.

What do you think should be the right choice here? Is lxml handling bom encoding fine?

For lxml it seems that it is converting back to bytes in line 101.